src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -9628,3 +9629,73 @@ void ObjectMgr::LoadSceneTemplates()
 
     TC_LOG_INFO("server.loading", ">> Loaded %u scene templates in %u ms.", count, GetMSTimeDiffToNow(oldMSTime));
 }
+
+void ObjectMgr::LoadAreaTriggerTemplates()
+{
+    uint32 oldMSTime = getMSTime();
+    _areaTriggerTemplateStore.clear();
+
+    PreparedStatement* stmt = WorldDatabase.GetPreparedStatement(WORLD_SEL_AREATRIGGER_TEMPLATES);
+    PreparedQueryResult templates = WorldDatabase.Query(stmt);
+
+    if (!templates)
+    {
+        TC_LOG_INFO("server.loading", ">> Loaded 0 AreaTrigger templates. DB table `areatrigger_template` is empty.");
+        return;
+    }
+
+    uint32 count = 1;
+
+    // We create a default template in case of the spell reference an unknown template
+    AreaTriggerTemplate defaultAreaTriggerTemplate;
+    defaultAreaTriggerTemplate.Id = 0;
+    defaultAreaTriggerTemplate.Flags = AREATRIGGER_FLAG_HAS_SPHERE;
+    defaultAreaTriggerTemplate.SphereDatas.Radius = 1;
+    defaultAreaTriggerTemplate.SphereDatas.RadiusTarget = 1;
+    defaultAreaTriggerTemplate.InitMaxSearchRadius();
+    _areaTriggerTemplateStore[0] = defaultAreaTriggerTemplate;
+
+    do
+    {
+        Field* fields = templates->Fetch();
+        AreaTriggerTemplate areaTriggerTemplate;
+        areaTriggerTemplate.Id = fields[0].GetUInt32();
+        areaTriggerTemplate.Flags = fields[1].GetUInt32();
+        areaTriggerTemplate.MoveCurveId = fields[2].GetInt32();
+        areaTriggerTemplate.ScaleCurveId = fields[3].GetInt32();
+        areaTriggerTemplate.MorphCurveId = fields[4].GetInt32();
+        areaTriggerTemplate.FacingCurveId = fields[5].GetInt32();
+
+        for (uint8 i = 0; i < MAX_AREATRIGGER_ENTITY_DATA; ++i)
+            areaTriggerTemplate.DefaultDatas.Data[i] = fields[6 + i].GetFloat();
+
+        areaTriggerTemplate.ScriptId = sObjectMgr->GetScriptId(fields[12].GetCString());
+
+        PreparedStatement* verticesStmt = WorldDatabase.GetPreparedStatement(WORLD_SEL_AREATRIGGER_POLYGON_VERTICES);
```

Comment:
Select vertices once at the start and cache them in map<areatriggerId, vector<vertice>>, then just std::move that vector when looping over result from templates